### ISSUE TITLE
Only allow specific compontent names

### ISSUE THREAD

**Issue by fisker** [vuejs/eslint-plugin-vue#2568]
**Please describe what the rule should do:**

Enforce consistency component name prefix. This rule should accept options to enforce component matches names or name patterns.
 
**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- Good -->
<keep-alive></keep-alive> <!-- builtin -->
<button/> <!-- html tag -->
<foo-button/> <!-- allowed with option `/^foo\-/` -->

<!-- Bad -->
<bar-button/>
```

**Additional context**

N/A

**Comment by ota-meshi** [comment]
Thank you for proposing the rule. That rule sounds good to me.
